From 47c7de68cdf12012f2004c0d563374f89f1bcba2 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Wed, 22 Jul 2015 12:35:32 -0700
Subject: [PATCH 1/2] text: properly block right click button-press-event

We used to destroy the menu as a workaround that we could not introspect
the event button.
Now that we can do that, just eat right click events.
This fixes a crash when right clicking with recent GtkSourceView
---
 src/js/viewers/text.js | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/src/js/viewers/text.js b/src/js/viewers/text.js
index 291d51b..2410ec9 100644
--- a/src/js/viewers/text.js
+++ b/src/js/viewers/text.js
@@ -23,6 +23,7 @@
  *
  */
 
+const Gdk = imports.gi.Gdk;
 const GtkClutter = imports.gi.GtkClutter;
 const Gtk = imports.gi.Gtk;
 const GLib = imports.gi.GLib;
@@ -84,13 +85,13 @@ const TextRenderer = new Lang.Class({
         if (this._buffer.get_language())
             this._view.set_show_line_numbers(true);
 
-        // FIXME: *very* ugly wokaround to the fact that we can't
-        // access event.button from a button-press callback to block
-        // right click
-        this._view.connect('populate-popup',
-                           Lang.bind(this, function(widget, menu) {
-                               menu.destroy();
-                           }));
+        this._view.connect('button-press-event', Lang.bind(this, function(view, event) {
+            let [, button] = event.get_button();
+            if (button == Gdk.BUTTON_SECONDARY)
+                return true;
+
+            return false;
+        }));
 
         this._scrolledWin = Gtk.ScrolledWindow.new(null, null);
         this._scrolledWin.add(this._view);
-- 
2.4.6

